libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxOnboardSpeechPlayer.kt

```diff
@@ -0,0 +1,129 @@
+package com.mapbox.navigation.ui.voice.api
+
+import android.content.Context
+import android.os.Bundle
+import android.speech.tts.TextToSpeech
+import android.speech.tts.UtteranceProgressListener
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayerCallback
+import com.mapbox.navigation.ui.base.model.voice.Announcement
+import com.mapbox.navigation.ui.base.model.voice.SpeechState
+import java.util.Locale
+
+/**
+ * Offline implementation of [SpeechPlayer].
+ * @property context Context
+ * @property language String
+ */
+internal class MapboxOnboardSpeechPlayer(
```

Comment:
Considering that this is an internal class, and it is only used by the MapboxHybridSpeechPlayer

Consider not making it implement the SpeechPayer. That way the interactions are specific for MapboxHybridSpeechPlayer.

For example, `MapboxOffboardSpeechPlayer` can only play a File. So it fails when you send it a `SpeechState.Play` without a `File`.

Think it can be more explicit: 
-- MapboxHybridSpeechPlayer.play(SpeechState.Play)
---- MapboxOffboardSpeechPlayer.play(file: File)

Reply:
It doesn't fail. Basically, it doesn't play 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/9d9590ccafcb89459af455a9fe70b84435ce81d4/libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxOffboardSpeechPlayer.kt#L43-L46

Actually, we're checking if the file exists or not in `MapboxHybridSpeechPlayer` to dispatch the playing to `MapboxOffboardSpeechPlayer` or `MapboxOnboardSpeechPlayer` https://github.com/mapbox/mapbox-navigation-android/blob/9d9590ccafcb89459af455a9fe70b84435ce81d4/libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxHybridSpeechPlayer.kt#L86-L88

Reply:
yeah this is what I meant by fail, is that it doesn't play. but looks like it is an internal class now so marking it resolved